### PR TITLE
Add Herringbone Pro, re-organize herringbone folder

### PR DESCRIPTION
**Pull request by ramonimbao**
## Description

Previous files from `ramonimbao/herringbone` were moved to `ramonimbao/herringbone/v1`. An additional `pro` folder was added for the newer version.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/ramonimbao/herringbone/pro/keymaps/via/keymap.c [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    if (clockwise) {
+        encoder_cw.pressed = true;
+        encoder_cw.time = (timer_read() | 1);
+        action_exec(encoder_cw);
+    } else {
+        encoder_ccw.pressed = true;
+        encoder_ccw.time = (timer_read() | 1);
+        action_exec(encoder_ccw);
+    }
+}
```
Comment (drashna) [inline keyboards/ramonimbao/herringbone/pro/keymaps/via/keymap.c:87]: Oh my. That's cheating.
LOVE IT!

**Review by drashna — APPROVED** [review]

File: keyboards/ramonimbao/herringbone/pro/keymaps/via/keymap.c (hunk 1 above)
Comment (ramonimbao) [reply keyboards/ramonimbao/herringbone/pro/keymaps/via/keymap.c:87]: Haha, can't take credit for this though. Not sure where this was first done, but I saw this here: https://github.com/NightlyBoards/qmk_firmware/commit/fe7949d6f5783c184f3e7d3ee4df7b4bb86f26ba

File: keyboards/ramonimbao/herringbone/pro/config.h [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+//#define MAGIC_KEY_LAYER8         8
+//#define MAGIC_KEY_LAYER9         9
+//#define MAGIC_KEY_BOOTLOADER     B
+//#define MAGIC_KEY_BOOTLOADER_ALT ESC
+//#define MAGIC_KEY_LOCK           CAPS
+//#define MAGIC_KEY_EEPROM         E
+//#define MAGIC_KEY_EEPROM_CLEAR   BSPACE
+//#define MAGIC_KEY_NKRO           N
+//#define MAGIC_KEY_SLEEP_LED      Z
+
```
Comment (fauxpark) [inline keyboards/ramonimbao/herringbone/pro/config.h:145]:
```suggestion
```

File: keyboards/ramonimbao/herringbone/pro/rules.mk [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+F_CPU = 16000000
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (fauxpark) [inline keyboards/ramonimbao/herringbone/pro/rules.mk:14]:
```suggestion
```

File: keyboards/ramonimbao/herringbone/pro/rules.mk [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+ENCODER_ENABLE = yes
```
Comment (fauxpark) [inline keyboards/ramonimbao/herringbone/pro/rules.mk:35]:
```suggestion
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
ENCODER_ENABLE = yes
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/ramonimbao/herringbone/pro/readme.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
+# Herringbone Pro
+
+![Herringbone Pro](hhttps://i.imgur.com/cewklY5.png)
```
Comment (noroadsleft) [inline keyboards/ramonimbao/herringbone/pro/readme.md:3]: Swaps this image for a smaller version.

```suggestion
![Herringbone Pro](https://i.imgur.com/cewklY5l.png)
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]
Also, can you add a file at `keyboards/ramonimbao/herringbone/rules.mk` with the following for the contents:

```makefile
DEFAULT_FOLDER = ramonimbao/herringbone/v1
```

This will make it so the previous `make` command still works.

**Comment by ramonimbao** [comment]
Currently implementing OLED support for this board.

File: keyboards/ramonimbao/herringbone/v1/rules.mk [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+F_CPU = 16000000
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (fauxpark) [inline keyboards/ramonimbao/herringbone/v1/rules.mk:14]:
```suggestion
```

File: keyboards/ramonimbao/herringbone/v1/rules.mk [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
```
Comment (fauxpark) [inline keyboards/ramonimbao/herringbone/v1/rules.mk:34]:
```suggestion
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/ramonimbao/herringbone/pro/rules.mk [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+ENCODER_ENABLE = yes
+OLED_DRIVER_ENABLE = yes
+WPM_ENABLE = yes
+
+SRC += pattern.c
+
+# Enable link-time optimization
+EXTRAFLAGS += -flto
```
Comment (fauxpark) [inline keyboards/ramonimbao/herringbone/pro/rules.mk:33]:
```suggestion
```

File: keyboards/ramonimbao/herringbone/pro/rules.mk [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+ENCODER_ENABLE = yes
+OLED_DRIVER_ENABLE = yes
+WPM_ENABLE = yes
```
Comment (fauxpark) [inline keyboards/ramonimbao/herringbone/pro/rules.mk:28]:
```suggestion
WPM_ENABLE = yes
LTO_ENABLE = yes
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/ramonimbao/herringbone/pro/readme.md [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+A through-hole 75% keyboard with a rotary encoder and OLED display!
+
+* Keyboard Maintainer: [Ramon Imbao](https://github.com/ramonimbao)
+* Hardware Supported: ATmega32A
+* Hardware Availability: [Herringbone](https://github.com/ramonimbao/Herringbone)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make ramonimbao/herringbone/pro:default
+
```
Comment (Erovia) [inline keyboards/ramonimbao/herringbone/pro/readme.md:14]: Can you add "clear instructions on how to reset the board into bootloader mode" as per [PR Checklist](https://docs.qmk.fm/#/pr_checklist?id=keyboard-prs)?

File: keyboards/ramonimbao/herringbone/v1/readme.md [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+A through-hole 75% keyboard.
+
+* Keyboard Maintainer: [Ramon Imbao](https://github.com/ramonimbao)
+* Hardware Supported: ATmega32A
+* Hardware Availability: [Herringbone](https://github.com/ramonimbao/Herringbone)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make ramonimbao/herringbone/v1:default
+
```
Comment (Erovia) [inline keyboards/ramonimbao/herringbone/v1/readme.md:14]: Can you add "clear instructions on how to reset the board into bootloader mode" as per [PR Checklist](https://docs.qmk.fm/#/pr_checklist?id=keyboard-prs)?

File: keyboards/ramonimbao/herringbone/readme.md [hunk 12]
```diff
@@ -1,14 +1,15 @@
...
-![Herringbone](https://i.imgur.com/SQyS0j8.png)
-
-A through-hole 75% keyboard.
+Through-hole 75% keyboards. The Pro version has support for rotary knob, OLED display, and split space.
 
 * Keyboard Maintainer: [Ramon Imbao](https://github.com/ramonimbao)
 * Hardware Supported: ATmega32A
+* Hardware Availability: [Herringbone](https://github.com/ramonimbao/Herringbone)
 
 Make example for this keyboard (after setting up your build environment):
```
Comment (Erovia) [inline keyboards/ramonimbao/herringbone/readme.md:9]:
```suggestion
Make example for these keyboards (after setting up your build environment):
```

File: keyboards/ramonimbao/herringbone/pro/info.json [hunk 13]
```diff
@@ -0,0 +1,321 @@
...
+                {"x": 3.75, "y": 5.25, "w": 6.25},
+                {"x": 10, "y": 5.25, "w": 1.5},
+                {"x": 11.5, "y": 5.25, "w": 1.5},
+
+                {"x": 13.25, "y": 5.5},
+                {"x": 14.25, "y": 5.5},
+                {"x": 15.25, "y": 5.5}
+            ]
+        },
+        "LAYOUT_all": {
```
Comment (Erovia) [inline keyboards/ramonimbao/herringbone/pro/info.json:212]: image
This is how `LAYOUT_all` renders, is this correct?

**Review by Erovia — CHANGES_REQUESTED** [review]

File: keyboards/ramonimbao/herringbone/pro/info.json (hunk 13 above)
Comment (ramonimbao) [reply keyboards/ramonimbao/herringbone/pro/info.json:212]: There should be a tiny bit of space between the last three keys. I fixed that now, thanks.

File: keyboards/ramonimbao/herringbone/pro/pro.h [hunk 14]
```diff
@@ -0,0 +1,87 @@
...
+    { k40, k41, k42, k43, k44, k45, k46, k47, k48, k49, k4a, k4b, XXX, k4d, k4e }, \
+    { k50, XXX, k52, k53, XXX, k55, XXX, XXX, XXX, k59, k5a, XXX, k5c, k5d, k5e }, \
+    { XXX, XXX, XXX, XXX, XXX, XXX, XXX, XXX, XXX, XXX, XXX, XXX, XXX, XXX, XXX} \
+}
+
+#define LAYOUT_iso( \
+    k00, k01, k02, k03, k04, k05, k06, k07, k08, k09, k0a, k0b, k0c, k0d, k0e, \
+    k10, k11, k12, k13, k14, k15, k16, k17, k18, k19, k1a, k1b, k1c, k1d, k1e, \
+    k20, k21, k22, k23, k24, k25, k26, k27, k28, k29, k2a, k2b, k2c, k2d, k2e, \
+    k30, k31, k32, k33, k34, k35, k36, k37, k38, k39, k3a, k3b,      k3d, k3e, \
```
Comment (noroadsleft) [inline keyboards/ramonimbao/herringbone/pro/pro.h:54]: QMK's custom is that Enter key keycodes should be toward the end of the home row.

```suggestion
    k20, k21, k22, k23, k24, k25, k26, k27, k28, k29, k2a, k2b, k2c,      k2e, \
    k30, k31, k32, k33, k34, k35, k36, k37, k38, k39, k3a, k3b, k3d, k2d, k3e, \
```

File: keyboards/ramonimbao/herringbone/pro/info.json [hunk 15]
```diff
@@ -0,0 +1,321 @@
...
+                {"x": 4.5, "y": 2.25},
+                {"x": 5.5, "y": 2.25},
+                {"x": 6.5, "y": 2.25},
+                {"x": 7.5, "y": 2.25},
+                {"x": 8.5, "y": 2.25},
+                {"x": 9.5, "y": 2.25},
+                {"x": 10.5, "y": 2.25},
+                {"x": 11.5, "y": 2.25},
+                {"x": 12.5, "y": 2.25},
+                {"x": 13.75, "y": 2.25, "w": 1.25, "h": 2},
```
Comment (noroadsleft) [inline keyboards/ramonimbao/herringbone/pro/info.json:163]: Moves the ISO Enter key's object to match the updated layout macro in `pro.h`.

(1/2)

```suggestion
```

File: keyboards/ramonimbao/herringbone/pro/info.json [hunk 16]
```diff
@@ -0,0 +1,321 @@
...
+                {"x": 4.75, "y": 3.25},
+                {"x": 5.75, "y": 3.25},
+                {"x": 6.75, "y": 3.25},
+                {"x": 7.75, "y": 3.25},
+                {"x": 8.75, "y": 3.25},
+                {"x": 9.75, "y": 3.25},
+                {"x": 10.75, "y": 3.25},
+                {"x": 11.75, "y": 3.25},
+                {"x": 12.75, "y": 3.25},
+                {"x": 15.5, "y": 3.25},
```
Comment (noroadsleft) [inline keyboards/ramonimbao/herringbone/pro/info.json:180]: (2/2)

```suggestion
                {"x": 13.75, "y": 2.25, "w": 1.25, "h": 2},
                {"x": 15.5, "y": 3.25},
```

File: keyboards/ramonimbao/herringbone/pro/keymaps/iso/keymap.c [hunk 17]
```diff
@@ -0,0 +1,98 @@
...
+#include QMK_KEYBOARD_H
+
+#include "pattern.h"
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT_iso(
+        KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_PSCR, KC_MUTE,
+        KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_HOME,
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_ENT,  KC_PGUP,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_NUHS,          KC_PGDN,
```
Comment (noroadsleft) [inline keyboards/ramonimbao/herringbone/pro/keymaps/iso/keymap.c:25]: And also update the keymaps to match the updated macro.

```suggestion
        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC,          KC_PGUP,
        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_NUHS, KC_ENT,  KC_PGDN,
```

File: keyboards/ramonimbao/herringbone/v1/readme.md [hunk 18]
```diff
@@ -0,0 +1,23 @@
+# Herringbone
+
+![Herringbone](https://i.imgur.com/SQyS0j8.png)
```
Comment (noroadsleft) [inline keyboards/ramonimbao/herringbone/v1/readme.md:3]: Swaps this image for a smaller version.

```suggestion
![Herringbone](https://i.imgur.com/SQyS0j8l.png)
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

File: keyboards/ramonimbao/herringbone/pro/keymaps/iso/keymap.c (hunk 17 above)
Comment (ramonimbao) [reply keyboards/ramonimbao/herringbone/pro/keymaps/iso/keymap.c]: Thanks. I'll take note of this convention for future layouts.

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**